src/insert.rs

```diff
@@ -128,9 +130,14 @@ impl<T> Insert<T> {
 
         // TODO: what about escaping a table name?
         // https://clickhouse.com/docs/en/sql-reference/syntax#identifiers
-        let sql = format!("INSERT INTO {table}({fields}) FORMAT RowBinary");
+        let format = if row_metadata.is_some() {
+            "RowBinaryWithNamesAndTypes"
```

Comment:
I'd like to see the format types defined as an `enum`, or at least constants somewhere. That's something I could do. If we want the user to be able to specify formats that we don't know about (e.g if new ones get added), there's options for that.

Reply:
Currently, this is internal only; we can add some popular formats as an enum provided by the crate. This might be useful with `fetch_bytes`: https://github.com/ClickHouse/clickhouse-rs/blob/main/examples/stream_arbitrary_format_rows.rs
and its INSERT counterpart that is to be implemented: https://github.com/ClickHouse/clickhouse-rs/issues/174

So perhaps it can be introduced in the scope of #174 indeed. However: it should implement `Into<String>` and not break the existing API expecting an `impl Into<String>`, cause we want to export these enums as just a _hint_ on what is available, and this library might not be in sync with the latest CH release that adds a new format, which the end user might want to use ASAP.